ui/components/component-library/select-wrapper/select-wrapper.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState, useRef } from 'react';
+import classnames from 'classnames';
+import type { PolymorphicRef, BoxProps } from '../box';
+import { Box, Popover, PopoverPosition } from '..';
+import {
+  SelectWrapperComponent,
+  SelectWrapperProps,
+} from './select-wrapper.types';
+import { SelectContext } from './select-wrapper.context';
+
+export const SelectWrapper: SelectWrapperComponent = React.forwardRef(
+  <C extends React.ElementType = 'div'>(
+    {
+      className = '',
+      defaultValue,
+      value,
+      onValueChange,
+      placeholder,
+      isDanger,
+      isDisabled,
+      isOpen,
+      onOpenChange,
+      isMultiSelect,
+      triggerComponent,
+      popoverProps,
+      children,
+      onBlur,
+      ...props
+    }: SelectWrapperProps<C>,
+    ref?: PolymorphicRef<C>,
+  ) => {
+    const [uncontrolledValue, setUncontrolledValue] = useState<any | null>();
+    const [isUncontrolledOpen, setIsUncontrolledOpen] =
+      useState<boolean>(false);
+    const [referenceElement, setReferenceElement] =
+      useState<HTMLElement | null>();
+    const popoverRef = useRef<HTMLDivElement | null>(null);
+    const wrapperRef = useRef<HTMLDivElement | null>(null);
+
+    const setBoxRef = (anchorRef: HTMLElement | null) => {
+      setReferenceElement(anchorRef);
+    };
+
+    const toggleUncontrolledOpen = () => {
+      setIsUncontrolledOpen(!isUncontrolledOpen);
+    };
+
+    const handleClickOutside = () => {
+      setIsUncontrolledOpen(false);
+      if (onOpenChange) {
+        onOpenChange(false);
+      }
```

Comment:
This is is essentially the where we handle the closing of the popover so we want to the onBlur to fire here if it exists 
```suggestion
      }
      if (onBlur) {
        onBlur();
      }
```